src/dmd/expressionsem.d

```diff
@@ -9788,7 +9788,9 @@ Expression semanticY(DotIdExp exp, Scope* sc, int flag)
     }
     else
     {
-        if (exp.e1.op == TOK.type || exp.e1.op == TOK.template_)
+        if (calledForAliasThis)
```

Comment:
It appears we first need and `AggregateDeclaration` and then it's `aliasthis` property.  Can't the `AggregateDeclaration` be determined from `exp` or `sc` to avoid having to pass in the `calledForAliasThis` parameter.  Looking at the other code in this function, it looks like we just need to find a way to obtain the `AggreegateDeclaration` from the existing parameters.

Reply:
I don't think the aggregate declaration can be obtained from the parameters. Consider that the DotIdExp passed to semanticY is the one which is constructed **with** the resolved aliasthis as it's `lhs` and the `sc` parameter represents the scope of the DotIdExp. The only part of the code which links the expression to the aggregate declaration is the one present in typesem.d where the aliasthis is resolved and then tried by calling semanticY

Reply:
Ok, then what about setting `flag` before calling `semanticY`?  And is there a constant defined somewhere so we don't have to use the magic number `1`?

Reply:
No there isn't. The dmd code that uses flag checks for it like this : `if (flag & 1)`, even the documentation says : `if flag == 1, stop error and return null` .